dotnet/src/SemanticKernel.Abstractions/PromptTemplate/PromptTemplateConfig.cs

```diff
@@ -169,7 +169,29 @@ internal IReadOnlyList<KernelParameterMetadata> GetKernelParametersMetadata()
     /// <exception cref="ArgumentException">Thrown when the deserialization returns null.</exception>
     public static PromptTemplateConfig FromJson(string json)
     {
-        var result = JsonSerializer.Deserialize<PromptTemplateConfig>(json, JsonOptionsCache.ReadPermissive);
-        return result ?? throw new ArgumentException("Unable to deserialize prompt template config from argument. The deserialization returned null.", nameof(json));
+        var promptTemplateConfig = JsonSerializer.Deserialize<PromptTemplateConfig>(json, JsonOptionsCache.ReadPermissive);
+        if (promptTemplateConfig == null)
+        {
+            throw new ArgumentException("Unable to deserialize prompt template config from argument. The deserialization returned null.", nameof(json));
+        }
+
+        // Prevent the default value from being any type other than a string.
+        foreach (var inputVariable in promptTemplateConfig.InputVariables)
+        {
+            // The value of the default property becomes a JsonElement after deserialization because that is how the JsonSerializer handles properties of the object type.
+            if (inputVariable.Default is JsonElement element)
+            {
+                if (element.ValueKind == JsonValueKind.String)
+                {
+                    inputVariable.Default = element.ToString();
+                }
+                else
+                {
+                    throw new ArgumentException($"Default value for input variable '{inputVariable.Name}' must be a string. Prompt function - '{promptTemplateConfig.Name ?? promptTemplateConfig.Description}'.");
```

Comment:
This is a temporary limitation so can the error message reflect that.

Reply:
NotSupportedException ?
```suggestion
                    throw new NotSupportedException($"Default value for input variable '{inputVariable.Name}' must be a string. Prompt function - '{promptTemplateConfig.Name ?? promptTemplateConfig.Description}'.");
```

Reply:
Done